bot.py

```diff
@@ -1714,6 +1712,41 @@ async def before_autoupdate(self):
             self.autoupdate.cancel()
             return
 
+    @tasks.loop()
+    async def log_expiry(self):
+        log_expire_after = self.config.get("log_expiration")
+        if log_expire_after == isodate.Duration():
+            return self.log_expiry.stop()
+
+        now = datetime.utcnow()
+        expiration_datetime = now - log_expire_after
+        expired_logs = await self.db.logs.find({"closed_at": {"$lte": str(expiration_datetime)}}).to_list(
+            None
+        )
+
+        if not await self.db.logs.find_one():
+            return await discord.utils.sleep_until(now + timedelta(days=1))
+
+        if not expired_logs:
+            fetch = await self.db.logs.find().sort("closed_at", 1).limit(1).to_list(None)
+            if fetch:
+                for x in fetch:
+                    if x["closed_at"] > str(expiration_datetime):
+                        closed_at = datetime.strptime(x["closed_at"], "%Y-%m-%d %H:%M:%S.%f%z")
+                        next_expiry = closed_at + log_expire_after
+                        return await discord.utils.sleep_until(next_expiry)
+
+        await self.db.logs.delete_many({"closed_at": {"$lte": str(expiration_datetime)}})
```

Comment:
`closed_at` is stored as a string, see #3115. You cannot directly compare `closed_at` using `lte` string comparison. You'll need to use an MongoDB aggregation pipeline function (most likely `$dateFromString`) then compare as date types.

Reply:
The string compare does work as its comparing both of them as strings. The issue there is a database integrity issue where we are storing date time as a string, this would require a overhaul of the existing databases